### DISCUSSION TITLE
Missing localization variables

### DISCUSSION THREAD

**Discussion by dd-mozdemir** [wixtoolset/issues#6627] [Questions] [answered]
Hey everyone,

I am trying to convert a project to WiX 4.0 and I'm running into an issue where I have errors about localization variables being missing when I include my own localization file. 

I pulled some random errors out of the output because there are a lot of them, bulk of these appear to come from the UI and Firewall extensions.  Here they are:

```
C:\projects\wix4\src\ext\UI\wixlib\Common.wxs(17) : error WIX0102: The localization variable !(loc.UITextbytes) is unknown.  Please ensure the variable is defined.
C:\projects\wix4\src\ext\UI\wixlib\ErrorProgressText.wxs(232) : error WIX0102: The localization variable !(loc.Error1929) is unknown.  Please ensure the variable is defined.
C:\projects\wix4\src\ext\UI\wixlib\ErrorProgressText.wxs(233) : error WIX0102: The localization variable !(loc.Error1930) is unknown.  Please ensure the variable is defined.
C:\projects\wix4\src\ext\Firewall\wixlib\FirewallExtension_Platform.wxi(10) : error WIX0102: The localization variable !(loc.WixRollbackFirewallExceptionsInstall) is unknown.  Please ensure the variable is defined.
C:\projects\wix4\src\ext\UI\wixlib\MsiRMFilesInUse.wxs(20) : error WIX0102: The localization variable !(loc.MsiRMFilesInUseDontUseRM) is unknown.  Please ensure the variable is defined.

**... (There's a lot more)**

```

When I include some of these variables that appear in the error message, like `UITextbytes` in my own localization file that particular error goes away.  My question is, do I need to include all of these localization variables that appear in the error messages in my `localization.wxl` file or am I missing something?  (An extension, a command-line parameter, etc)

The command I'm running looks like this:  

```
wix build source.wxs -ext WixToolset.Bal.wixext.dll -ext WixToolset.Util.wixext.dll -ext WixToolset.Firewall.wixext.dll -ext WixToolset.UI.wixext.dll -loc localization-en-us.wxl
```

**Comment by dd-mozdemir** [comment, marked as answer]
nevermind on this, I did convert the localization file but it did not add the required WixLocalization section to the file.  I looked at this [file ](https://github.com/wixtoolset/wix4/blob/fad58e4d2bbd8e451cdb725325a5d760b4d1a684/src/ext/UI/wixlib/WixUI_en-us.wxl) and changed my the heading of my localization file to:

`<WixLocalization Culture="en-US" Codepage="1252" xmlns="http://wixtoolset.org/schemas/v4/wxl">` 

This made the issue go away.

**Reply by robmen** [reply]
Added https://github.com/wixtoolset/issues/issues/6639 to track the lack of error message.
